### PR TITLE
Fix inputs of invalid phone numbers

### LINKED ISSUE #156: [PE-D][Tester C] Extremely long invalid phone numbers can be successfully created for a client profile
When adding a client profile, I can create a client who has an extremely long phone number like 1234567890987654321234567890987654321, which is an invalid phone number but it can be created successfully as shown in the screenshot below.

Perhaps apart from limiting to phone numbers having at least 3 digits, can add restrictions  to check if the phone number exceeds 8 digits?

-------------
Labels: `severity.Medium` `type.FeatureFlaw`
original: Jweng88/ped#4

### PR DESCRIPTION
**Pull request by MagnificentCreature**
Fix #156 and #158

### CODE REVIEW

File: src/test/java/seedu/address/model/person/PhoneTest.java [hunk 1]
```diff
@@ -28,14 +28,20 @@ public void isValidPhone() {
         assertFalse(Phone.isValidPhone("")); // empty string
         assertFalse(Phone.isValidPhone(" ")); // spaces only
         assertFalse(Phone.isValidPhone("91")); // less than 3 numbers
+        assertFalse(Phone.isValidPhone("+91")); // less than 3 numbers with plus
         assertFalse(Phone.isValidPhone("phone")); // non-numeric
         assertFalse(Phone.isValidPhone("9011p041")); // alphabets within digits
+        assertFalse(Phone.isValidPhone("9312-1534")); //other characters
+        assertFalse(Phone.isValidPhone("123456789012345678901234567890")); // too long phone numbers with plus sign
         assertFalse(Phone.isValidPhone("9312 1534")); // spaces within digits
```
Comment (zhyuhan) [inline src/test/java/seedu/address/model/person/PhoneTest.java:36]: Shall we test some boundary values? Like a phone number with 16 numbers, and 16 numbers with a `+`?

File: src/main/java/seedu/address/model/person/Phone.java [hunk 2]
```diff
@@ -11,8 +11,9 @@ public class Phone {
 
 
     public static final String MESSAGE_CONSTRAINTS =
-            "Phone numbers should only contain numbers, and it should be at least 3 digits long";
-    public static final String VALIDATION_REGEX = "\\d{3,}";
+        "Phone numbers should only contain numbers (with an optional + sign at the front)"
+        + "and they should be between 3 and 15 characters long";
```
Comment (zhyuhan) [inline src/main/java/seedu/address/model/person/Phone.java]: This can be interpreted as 15 characters including the optional `+`. Maybe we can move the `(with an optional + sign at the front)` part to the back?

**Review by zhyuhan — APPROVED** [review]

File: src/test/java/seedu/address/model/person/PhoneTest.java (hunk 1 above)
Comment (MagnificentCreature) [reply src/test/java/seedu/address/model/person/PhoneTest.java:36]: That is already done in line 44

**Review by KamiliArsyad — APPROVED** [review]
Might be better to have invalid number generator for the testcases or something but otherwise oke LGTM
